src/gf_ring.h

```diff
@@ -148,12 +148,13 @@ class RingModN {
     std::vector<int> exponents;
     std::vector<T> all_primes_factors;
     std::vector<T> proper_divisors;
+    // random number engine with a seed as a random number
+    mutable std::default_random_engine generator{std::random_device()()};
+    mutable std::uniform_int_distribution<T> distribution;
```

Comment:
You can't use `T`, 128-bit is not standard and thus not supported.

Also why mutable? And the init should probably be done in the constructor initialization list, not here.

Reply:
> You can't use `T`, 128-bit is not standard and thus not supported.

Yep, we can't use `T` here :(

> Also why mutable? 

I set mutable as `distribution` and `generator` will be called from a const instance of GF, while their state will be updated after each call.

> And the init should probably be done in the constructor initialization list, not here.

Yes, they will be updated :)

Reply:
can you use PRNG() ?

Reply:
it's done